dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -139,17 +143,26 @@ public String route(final ContainerRequestContext request) {
             Class<?> resourceClass = (Class<?>) request.getProperty(resourceString + "class");
             Method method = (Method) request.getProperty(resourceString + "method");
 
-            UriBuilder template = UriBuilder.fromResource(resourceClass);
-            String contextRoot = request.getUriInfo().getBaseUri().getPath();
-            if (contextRoot != null) {
-                template.path(contextRoot);
-            }
+            String key = resourceClass.getName() + "." + method.getName();
```

Comment:
I have a feeling that this won't work if you have the same class deployed on two different applications in the same server with different context roots.  The cache code doesn't take into account the context root.  Also the method name is not unique enough.  You can have the method name multiple times with different parameters on the same class.  As such you end up with issues returning the wrong route when each method has a different @Path.  I think you can make this work, but you will still need to get the context root and method if there is a @Path.  

The code also isn't great for when the method doesn't have @Path.  It should also check to see if the Class has @Path if the method doesn't.  That isn't from your code change.  It is just wrong in general.  

As to caching perspective and making sure you don't end up leaking, you could end up with a WeakReference as your key that points to the Class and have fields on the reference for contextRoot and possibly the Method.  I don't think that Method will keep the Class in memory though, but would have to try it out.  With that you would be able to cache correctly and then clear out the weak references when an application is stopped for instance.

Reply:
I am wrong about the @Path on the class.  The fromResource takes care of that.

Reply:
I really wonder what this route looks like because it does seem odd with the contextRoot.